components/DetectAnimatedGif.coffee

```diff
@@ -22,7 +22,27 @@ exports.getComponent = ->
     async: true
     forwardGroups: true
   , (buffer, groups, out, callback) ->
-    out.send animated buffer
-    do callback
+    if Buffer.isBuffer buffer
+      out.send animated buffer
+      do callback
+      return
+    else if typeof buffer is 'string'
+      isAnimated = false
+      fs.createReadStream buffer
+        .pipe animated()
+        .once 'animated', ->
+          isAnimated = true
+          out.send true
+          do callback
+          return
+        .on 'finish', ->
+          unless isAnimated
```

Comment:
The check is not the problem. What I am saying is that I think sending should only happen in finish.

So remove `out.send true\ndo callback` from `once 'animated'` and then in `finish` do `out.send isAnimated\n do callback`